jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/PerProcessInstanceRuntimeManager.java

```diff
@@ -182,18 +182,23 @@ public void signalEvent(String type, Object event) {
         }
     
         // next find out all instances waiting for given event type
+        runtimeEngine = null;
         List<String> processInstances = ((InternalMapper) mapper).findContextIdForEvent(type, getIdentifier());
         for (String piId : processInstances) {
-            runtimeEngine = getRuntimeEngine(ProcessInstanceIdContext.get(Long.parseLong(piId)));
             try {
+                runtimeEngine = (RuntimeEngineImpl) getRuntimeEngine(ProcessInstanceIdContext.get(Long.parseLong(piId)));
                 // signal execution can rise errors
                 if (!signalledEngines.contains(runtimeEngine)) {
-                    runtimeEngine.getKieSession().signalEvent(type, event);
                     signalledEngines.add(runtimeEngine);
+                    runtimeEngine.getKieSession().signalEvent(type, event);
                 }
+            } catch (org.drools.persistence.api.SessionNotFoundException ex) {
+                logger.warn("Signal event cannot proceed because of session not found exception {} for engine {}", ex.getMessage(), runtimeEngine.getKieSessionId());
```

Comment:
I see this getting logged frequently and think it would be better to have it at DEBUG level.

Reply:
set to debug